nat-handler.py

```diff
@@ -94,17 +102,30 @@ def current_az(self):
     def current_region(self):
         return self.current_az[:-1]
 
+    @property
+    def primary_interface_id(self):
+        ec2 = connect_to_ec2(self.current_region)
+        instance = ec2.get_only_instances(instance_ids=[self.current_instance_id])[0]
+        interfaces = instance.interfaces
+        interface_id = [i.id for i in interfaces if i.id != self.secondary_interface_id]
+        return interface_id[0]
+
+    @property
+    def secondary_interface_id(self):
+        return self.config.secondary_interface_id(self.current_az)
+
     def take_route(self, az):
         route_table_id = self.config.route_table_id(az)
         vpc = connect_to_vpc(self.current_region)
-        vpc.replace_route(route_table_id, '0.0.0.0/0', instance_id=self.current_instance_id)
+        vpc.replace_route(route_table_id, '0.0.0.0/0', interface_id=self.primary_interface_id)
 
     def take_elastic_ip(self, az):
+        interface_id = self.primary_interface_id if az == self.current_az else self.secondary_interface_id
```

Comment:
This requires some detailed explanation, so just to recap:

For this case, it would mean that there is a failover and this was triggered on serf's member-leave/ member-failed event where an instance was trying to associate an EIP of another availability zone (AZ) for which the nat instance serving that AZ failed.

So in the ideal case, the instance taking over the EIP would have the secondary interface (eth1) free and therefore associate the EIP to eth1.

If eth1 already has an EIP, it means it participated in a failover before and another NAT instance has not yet come up or is not functional for that AZ, otherwise it would reclaim the EIP. This also means that this case should never happen since we don't have quorum (at this point atleast two instances are not functional). Would you agree?

But if eth1 is already associated with an EIP, it will now be associated with the EIP of the given AZ. The EIP that was associated before with eth1 will not have any associations until an instance for that AZ can join the cluster can reclaim the EIP.

Do you see something wrong with this scenario?